imvertorlite.py

```diff
@@ -1,137 +1,189 @@
-    convert()
+import json
+import sys
+import click
+from bs4 import BeautifulSoup, ResultSet, Tag
+from utils import type_conversion, type_convert_dictionary
+import cchardet
+
+
+def get_initial_value(element: Tag):
+    initial = element.select_one("initial").get("body")
+    if initial:
+        return initial
+    else:
+        return None
+
+
+def extract_properties(attr: Tag):
+    attr_name = attr.get("name")
+    attr_dict = {
+        "$id": f"#/properties/{attr_name}",
+        "title": f"{attr_name}",
+    }
+
+    attr_props = attr.select_one("properties")
+    attr_dict.update(
+        type_convert_dictionary(attr_props.attrs))
+
+    initial_value = get_initial_value(attr)
+    if initial_value:
+        attr_dict["default"] = type_conversion(initial_value)
+
+    attr_type = attr_props.get("type")
+    if attr_type:
+        attr_dict["type"] = type_conversion(attr_type)
+
+    stereotype = attr.select_one("stereotype")
+    if stereotype:
+        attr_dict["stereotype"] = stereotype.get("stereotype")
+
+    documentation = attr.select_one("documentation")
+    if documentation:
+        attr_dict["description"] = documentation.get("value")
+
+    tags = attr.select_one("tags").select("tag")
+    if tags:
+        attr_dict["tags"] = [{"name": tag.get("name"),
+                              "value": tag.get("value"),
+                              "notes": tag.get("notes")} for tag in tags]
+
+    return attr_dict
+
+
+def generate_schema(soup: BeautifulSoup, base: BeautifulSoup, attrs: ResultSet):
+    """
+    Converts a specific UML class to a JSON schema.
+
+    @param soup: The base soup of the XML file to resolve references.
+    @param base: The base UML class (BeautifulSoup object).
+    @param attrs: A ResultSet (list of elements) containing all the individual properties.
+    """
+
+    # Create a base schema.
+    schema = {
+        "$schema": "http://json-schema.org/draft-07/schema",
+        "$id": "https://objecttypes.vng.cloud/schema.json",
+        "title": base.get("name"),
+        # Properties are added below.
+        "properties": {},
+        # By default, don't allow for additional properties.
+        "additionalProperties": False,
+        "type": "object",
+    }
+    schema.update(
+        type_convert_dictionary(base.select_one("properties").attrs)
+    )
+    schema.update(
+        type_convert_dictionary(base.select_one("project").attrs)
+    )
+
+    schema['description'] = schema.pop('documentation')
+
+    for attr in attrs:
+        attr_dict = extract_properties(attr)
+
+        attr_name = attr_dict.get("title")
+        schema["properties"][attr_name] = attr_dict
+
+        if attr_dict.get("stereotype") == "enum":
+            enum = soup.select_one(f'element[name="enum_{attr_name}"]')
+            if enum:
+                enum_list = list(set([a.get('name') for a in enum.select("attribute")]))
+                schema["properties"][attr_name]["examples"] = [enum_list[0]]

```

Comment:
I think we can remove this on property-level, and add an example at the top level, based on `enum_list[0]` for enums, a random string for string properties, random numer for number properties